monoio/src/driver/op/read.rs

```diff
@@ -102,20 +105,29 @@ impl<T: IoBufMut> OpAble for Read<T> {
 
     #[cfg(all(any(feature = "legacy", feature = "poll-io"), windows))]
     fn legacy_call(&mut self) -> io::Result<u32> {
-        let fd = self.fd.as_raw_socket();
+        let fd = self.fd.raw_handle();
         let seek_offset = libc::off_t::try_from(self.offset)
             .map_err(|_| io::Error::new(io::ErrorKind::Other, "offset too big"))?;
-        syscall!(
-            recv(
+        let mut bytes_read = 0;
+        let ret = unsafe {
+            let mut overlapped: OVERLAPPED = std::mem::zeroed();
+            overlapped.Anonymous.Anonymous.Offset = (seek_offset as i64 & 0xFFFFFFFF) as u32;
+            overlapped.Anonymous.Anonymous.OffsetHigh =
+                ((seek_offset as i64 >> 32) & 0xFFFFFFFF) as u32;
+            // see https://learn.microsoft.com/zh-cn/windows/win32/api/fileapi/nf-fileapi-readfile
+            ReadFile(
```

Comment:
@ihciah notice this change, I mistakenly thought that this method was only used for network IO before.